### DISCUSSION TITLE
invoke send MediatR method from command handler and enent handler

### DISCUSSION THREAD

**Discussion by RJR-Robert-Kewei-Zhang** [LuckyPennySoftware/MediatR#756] [Q&A]
Hi @jbogard 

I am using MediatR in my ASP.NET Core application, I wander if the following implementation are right ways to use MediatR command handler:
1.  Invoke send method from a command handler to start another command handler.
2. Invoke send method from a event handler to start another command handler.

Thanks a lot,
Robert

**Comment by jbogard** [comment]
I try not to do this because it can make things more confusing to follow. I like to have ONE request handler per request, but you can have multiple notification handlers. This is what our teams have settled on at least.

**Reply by RJR-Robert-Kewei-Zhang** [reply]
Thanks Jimmy. So that means don't create any new query/command handler inside a query/command handler, instead, publish a notification and use notification handler to process the logic, is my understanding correct?

Thanks,
Robert

**Reply by jbogard** [reply]
Well, notifications are very different than commands semantically. I personally use notifications (and by proxy, domain events) very sparingly. First see if you can simply call another method on another class. That's a bit more obvious. But it's highly dependent on the nature of the two operations, there is no one answer here.

**Reply by RJR-Robert-Kewei-Zhang** [reply]
Thanks a lot Jimmy for your comment, it help me a lot.
Robert
